src/Storages/HDFS/StorageHDFS.cpp

```diff
@@ -712,6 +712,8 @@ void registerStorageHDFS(StorageFactory & factory)
         engine_args[0] = evaluateConstantExpressionOrIdentifierAsLiteral(engine_args[0], args.getLocalContext());
 
         String url = checkAndGetLiteralArgument<String>(engine_args[0], "url");
+        if (url.empty())
```

Comment:
this will not fix the error you were referring to in splitting the url. Let's also add validation like 
```
auto pos = url.find("//");
check (pos != std::string::npos && std::advance(pos, 2) != url.end())
pos = find('/', pos + 2);
check pos != std::string::npos
```
(but it should be better put in constructor of the hdfs buffer)